### ISSUE TITLE
Repeated logs in local mode

### ISSUE THREAD

**Issue by rahul-thakoor** [balena-os/balena-supervisor#1739]
Hi , I can see duplicate logs even when pushing for first time

```
root@support-check-dbus# npx balena push 192.168.100.14 -c
[Info]    Starting build on device 192.168.100.14
[Build]   [ntp-check] Step 1/6 : FROM balenalib/armv7hf-alpine
[Build]   [ntp-check]  ---> e1181b610748
[Build]   [ntp-check] Step 2/6 : RUN install_packages dbus
[Build]   [ntp-check]  ---> Running in 32ca84b13616
[Build]   [ntp-check] Here are a few details about this Docker image (For more information please visit https://www.balena.io/docs/reference/base-images/base-images/): 
[Build]   Architecture: ARM v7 
[Build]   OS: Alpine Linux 3.13 
[Build]   Variant: run variant 
[Build]   Default variable(s): UDEV=off 
[Build]   Extra features: 
[Build]   - Easy way to install packages with `install_packages <package-name>` command 
[Build]   - Run anywhere with cross-build feature  (for ARM only) 
[Build]   - Keep the container idling with `balena-idle` command 
[Build]   - Show base image details with `balena-info` command
[Build]   [ntp-check] fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/main/armv7/APKINDEX.tar.gz
[Build]   [ntp-check] fetch https://dl-cdn.alpinelinux.org/alpine/v3.13/community/armv7/APKINDEX.tar.gz
[Build]   [ntp-check] OK: 23 MiB in 62 packages
[Build]   [ntp-check] Removing intermediate container 32ca84b13616
[Build]   [ntp-check]  ---> 472e7c27a5e1
[Build]   [ntp-check] Step 3/6 : COPY start.sh ./
[Build]   [ntp-check]  ---> 43368e918559
[Build]   [ntp-check] Step 4/6 : CMD ["bash" , "./start.sh"]
[Build]   [ntp-check]  ---> Running in 4e9b2fb8ebf0
[Build]   [ntp-check] Removing intermediate container 4e9b2fb8ebf0
[Build]   [ntp-check]  ---> 28dabf891b13
[Build]   [ntp-check] Step 5/6 : LABEL io.resin.local.image=1
[Build]   [ntp-check]  ---> Running in 72df171338fc
[Build]   [ntp-check] Removing intermediate container 72df171338fc
[Build]   [ntp-check]  ---> 080d17060f07
[Build]   [ntp-check] Step 6/6 : LABEL io.resin.local.service=ntp-check
[Build]   [ntp-check]  ---> Running in 9f0a1e5a1239
[Build]   [ntp-check] Removing intermediate container 9f0a1e5a1239
[Build]   [ntp-check]  ---> 4ac2eb3619d4
[Build]   [ntp-check] Successfully built 4ac2eb3619d4
[Build]   [ntp-check] Successfully tagged local_image_ntp-check:latest

[Info]    Streaming device logs...
[Live]    Watching for file changes...
[Live]    Waiting for device state to settle...
[Logs]    [7/2/2021, 4:23:23 PM] Killing service 'ntp-check sha256:efc7b02de798d8dd2f52e80b82fd3aaaf9d6a176adcac44806febe001988efde'
[Logs]    [7/2/2021, 4:23:35 PM] Service exited 'ntp-check sha256:efc7b02de798d8dd2f52e80b82fd3aaaf9d6a176adcac44806febe001988efde'
[Logs]    [7/2/2021, 4:23:35 PM] Killed service 'ntp-check sha256:efc7b02de798d8dd2f52e80b82fd3aaaf9d6a176adcac44806febe001988efde'
[Logs]    [7/2/2021, 4:23:36 PM] Installing service 'ntp-check sha256:4ac2eb3619d4008fb6bbb64cf99bb8c137e898a82ef49543a1405b9a3de736e6'
[Logs]    [7/2/2021, 4:23:36 PM] Installed service 'ntp-check sha256:4ac2eb3619d4008fb6bbb64cf99bb8c137e898a82ef49543a1405b9a3de736e6'
[Logs]    [7/2/2021, 4:23:36 PM] Starting service 'ntp-check sha256:4ac2eb3619d4008fb6bbb64cf99bb8c137e898a82ef49543a1405b9a3de736e6'
[Logs]    [7/2/2021, 4:23:38 PM] Started service 'ntp-check sha256:4ac2eb3619d4008fb6bbb64cf99bb8c137e898a82ef49543a1405b9a3de736e6'
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check] Checking if device time is NTP synchronized
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check] method return time=1625228618.857159 sender=:1.1141 -> destination=:1.1138 serial=3 reply_serial=2
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]    array [
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       dict entry(
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          string "Timezone"
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          variant             string ""
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       )
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       dict entry(
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          string "LocalRTC"
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          variant             boolean false
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       )
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       dict entry(
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          string "CanNTP"
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          variant             boolean false
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       )
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       dict entry(
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          string "NTP"
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          variant             boolean false
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       )
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       dict entry(
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          string "NTPSynchronized"
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          variant             boolean true
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       )
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       dict entry(
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          string "TimeUSec"
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          variant             uint64 1625228618856007
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       )
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       dict entry(
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          string "RTCTimeUSec"
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          variant             uint64 0
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       )
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]    ]
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check] Idling...
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check] method return time=1625228618.857159 sender=:1.1141 -> destination=:1.1138 serial=3 reply_serial=2
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]    array [
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       dict entry(
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          string "Timezone"
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          variant             string ""
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       )
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       dict entry(
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          string "LocalRTC"
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          variant             boolean false
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       )
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       dict entry(
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          string "CanNTP"
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          variant             boolean false
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       )
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       dict entry(
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          string "NTP"
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          variant             boolean false
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       )
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       dict entry(
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          string "NTPSynchronized"
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          variant             boolean true
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       )
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       dict entry(
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          string "TimeUSec"
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          variant             uint64 1625228618856007
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       )
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       dict entry(
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          string "RTCTimeUSec"
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]          variant             uint64 0
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]       )
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check]    ]
[Logs]    [7/2/2021, 4:23:38 PM] [ntp-check] Idling...
[Live]    Device state settled

**Comment by cywang117** [comment]
I'm not sure which logs are supposed to be repeated. The NTP logs? For the most part local mode logs entries are the exact same as they would be in journal logs if the device is not in local mode, and I don't recall seeing any repetitions in logs I've investigated, Could you clarify @rahul-thakoor?

**Comment by rahul-thakoor** [comment]
the ` method return time=1625228618.857159 sender=:1.1141 -> destination=:1.1138 serial=3 reply_serial=2...` part should be logged once only and the container goes to idle state @cywang117

**Comment by 20k-ultra** [comment]
Have you seen this with other projects as well ?

**Comment by rahul-thakoor** [comment]
Yes. I usually see old logs or repeated logs. I will try to reproduce with another project and report

**Comment by rahul-thakoor** [comment]
```
root@openwrt# npx balena push 192.168.100.25
[Info]    Starting build on device 192.168.100.25
[Info]    No "docker-compose.yml" file found at "."
[Info]    Creating default composition with source: "."
[Build]   [main] Step 1/9 : FROM openwrtorg/rootfs:armvirt-64-19.07.7 as builder
[Build]   [main]  ---> b495a3da1395
[Build]   [main] Step 2/9 : RUN mkdir -p /var/lock
[Build]   [main]  ---> Using cache
[Build]   [main]  ---> 2d2e8f68bf9f
[Build]   [main] Step 3/9 : RUN opkg update &&     opkg install luci     wpad-wolfssl     iw-full     ip-full     kmod-mac80211     dnsmasq     iptables-mod-checksum
[Build]   [main]  ---> Using cache
[Build]   [main]  ---> 01bad4cf9878
[Build]   [main] Step 4/9 : WORKDIR /usr/src/app
[Build]   [main]  ---> Running in 26e9907defad
[Build]   [main] Removing intermediate container 26e9907defad
[Build]   [main]  ---> 6be98e58f75c
[Build]   [main] Step 5/9 : COPY start.sh ./
[Build]   [main]  ---> 8ca640116f79
[Build]   [main] Step 6/9 : RUN chmod +x start.sh
[Build]   [main]  ---> Running in c9aa9deb4f9a
[Build]   [main] Removing intermediate container c9aa9deb4f9a
[Build]   [main]  ---> 95f82c4a40db
[Build]   [main] Step 7/9 : CMD ["sh", "./start.sh"]
[Build]   [main]  ---> Running in 944248f7bcf1
[Build]   [main] Removing intermediate container 944248f7bcf1
[Build]   [main]  ---> 1bb5f4e2dc95
[Build]   [main] Step 8/9 : LABEL io.resin.local.image=1
[Build]   [main]  ---> Running in 7da7ccfaceb5
[Build]   [main] Removing intermediate container 7da7ccfaceb5
[Build]   [main]  ---> d4133db8fa99
[Build]   [main] Step 9/9 : LABEL io.resin.local.service=main
[Build]   [main]  ---> Running in fb6e698cbd91
[Build]   [main] Removing intermediate container fb6e698cbd91
[Build]   [main]  ---> fb44a8218490
[Build]   [main] Successfully built fb44a8218490
[Build]   [main] Successfully tagged local_image_main:latest

[Info]    Streaming device logs...
[Live]    Watching for file changes...
[Live]    Waiting for device state to settle...
[Logs]    [7/8/2021, 1:29:03 PM] Killing service 'main sha256:48e82014e1658014614df741f6dcb0a6809b7913674ce1e40d65c55ae4a3f21a'
[Logs]    [7/8/2021, 1:29:14 PM] Service exited 'main sha256:48e82014e1658014614df741f6dcb0a6809b7913674ce1e40d65c55ae4a3f21a'
[Logs]    [7/8/2021, 1:29:14 PM] Killed service 'main sha256:48e82014e1658014614df741f6dcb0a6809b7913674ce1e40d65c55ae4a3f21a'
[Logs]    [7/8/2021, 1:29:15 PM] Installing service 'main sha256:fb44a8218490da775b92c231675a009a9774540d7c8129869a3f894c2c0fb563'
[Logs]    [7/8/2021, 1:29:16 PM] Installed service 'main sha256:fb44a8218490da775b92c231675a009a9774540d7c8129869a3f894c2c0fb563'
[Logs]    [7/8/2021, 1:29:16 PM] Starting service 'main sha256:fb44a8218490da775b92c231675a009a9774540d7c8129869a3f894c2c0fb563'
[Logs]    [7/8/2021, 1:29:16 PM] Started service 'main sha256:fb44a8218490da775b92c231675a009a9774540d7c8129869a3f894c2c0fb563'
[Logs]    [7/8/2021, 1:29:16 PM] [main] Starting ...
[Logs]    [7/8/2021, 1:29:16 PM] [main] Starting ...

**Comment by 20k-ultra** [comment]
what are the versions of your hostOS and Supervisor ? I'll try to reproduce as well. It seems like pushing any project in localmode should reproduce the duplicate logs..

**Comment by rahul-thakoor** [comment]
`balenaOS 2.80.3+rev1`
`12.7.0`

**Comment by cywang117** [comment]
Thanks Rahul! The repeated local mode logs is probably because there are mistakenly multiple global listeners attached:
https://github.com/balena-io/balena-supervisor/blob/758f3caa48a53c00dd3230fe97e2f704968c4eb6/src/logging/local-backend.ts#L45-L51

**Comment by pipex** [comment]
Should be fixed by #2351
